### PR TITLE
feat: icp two decimals and zero as zero

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

Display more readable ICP values

# Rules

- If ICP is zero then 0 should be displayed - i.e. without decimals
- ICP with round number (12.0) should be displayed 12.00
- ICP should be displayed with max. 2 decimals (12.1 → 12.10, 12.12353 → 12.12, 12.00003 → 12.00) in Accounts, but with up to 8 decimals without tailing 0s in transaction details.
- However, if ICP value is lower than 0.01 then it should be as it is without tailing 0s up to 8 decimals (e.g., 0.000003 is displayed as 0.000003)

# Worth To Note

If an account balance is `12.004` ICP, `12.00` will be displayed. Converted means `98.52$` instead of `98.55$` at current rate.

# Changes

- refactor `formatICP` interface
- drop `formatNumberToString`
-implement logic
- remove explicit formatting for `formattedTransactionFeeICP` as it will inherits the rules
- introduce `detailed` option to display ICP with 8 decimals
- use `detailed` into transaction review and card

# Screenshots

Capture d’écran 2022-06-02 à 18 36 22
Capture d’écran 2022-06-02 à 18 36 32
Capture d’écran 2022-06-02 à 18 36 42
Capture d’écran 2022-06-02 à 18 36 44
Capture d’écran 2022-06-02 à 18 36 50
Capture d’écran 2022-06-02 à 18 36 55
Capture d’écran 2022-06-02 à 18 36 59
Capture d’écran 2022-06-02 à 18 37 57
Capture d’écran 2022-06-02 à 18 38 44

### CODE REVIEW

File: frontend/svelte/src/tests/lib/utils/icp.utils.spec.ts [hunk 1]
```diff
@@ -12,18 +12,21 @@ import {
...
-    expect(formatICP(BigInt(1000000000))).toEqual("10.00000000");
-    expect(formatICP(BigInt(1010000000))).toEqual("10.10000000");
-    expect(formatICP(BigInt(1012300000))).toEqual("10.12300000");
-    expect(formatICP(BigInt(20000000000))).toEqual("200.00000000");
-    expect(formatICP(BigInt(20000000001))).toEqual("200.00000001");
-    expect(formatICP(BigInt(200000000000))).toEqual(`2${"\u202F"}000.00000000`);
-    expect(formatICP(BigInt(200000000000000))).toEqual(
-      `2${"\u202F"}000${"\u202F"}000.00000000`
+    expect(formatICP({ value: BigInt(0) })).toEqual("0");
+    // TODO: this following test used to equals 0.0000001 but because of the new ICP conversion it now renders 0.00
```
Comment (lmuntaner) [inline frontend/svelte/src/tests/lib/utils/icp.utils.spec.ts:16]: Add a todo for tests with `detailed` param as well.

**Review by lmuntaner — APPROVED** [review]
Minor comment. Looks good to me, I guess you looked in all the calls to `formatICP`.

File: frontend/svelte/src/lib/utils/icp.utils.ts [hunk 2]
```diff
@@ -1,37 +1,60 @@
...
+
+  const converted: number = Number(value) / E8S_PER_ICP;
+  const decimals: number =
+    converted < 0.01
+      ? Math.max(countDecimals(converted), ICP_DISPLAYED_DECIMALS)
+      : detailed
+      ? Math.min(countDecimals(converted), ICP_DISPLAYED_DECIMALS_DETAILED)
+      : ICP_DISPLAYED_DECIMALS;
+
+  return new Intl.NumberFormat("fr-FR", {
```
Comment (lmuntaner) [inline frontend/svelte/src/lib/utils/icp.utils.ts:42]: Mabybe moving to `en-US` to get the thousands sign?

File: frontend/svelte/src/lib/utils/icp.utils.ts [hunk 3]
```diff
@@ -1,37 +1,60 @@
...
+    minimumFractionDigits: decimals,
+    maximumFractionDigits: decimals,
   })
     .format(Number(value) / E8S_PER_ICP)
```
[the remark this answers is not in the document]
Comment (peterpeterparker) [reply frontend/svelte/src/lib/utils/icp.utils.ts]: done, thanks!

File: frontend/svelte/src/lib/utils/icp.utils.ts [hunk 4]
```diff
@@ -1,37 +1,60 @@
...
-    maximumFractionDigits: fractionDigits,
+  detailed?: boolean;
+}): string => {
+  if (value === BigInt(0)) {
+    return "0";
+  }
+
+  const converted: number = Number(value) / E8S_PER_ICP;
+  const decimals: number =
+    converted < 0.01
```
[the remark this answers is not in the document]
Comment (peterpeterparker) [reply frontend/svelte/src/lib/utils/icp.utils.ts:36]: I thought about making a constant but I don't even now how to call it 😁

File: frontend/svelte/src/tests/lib/utils/icp.utils.spec.ts (hunk 1 above)
Comment (peterpeterparker) [reply frontend/svelte/src/tests/lib/utils/icp.utils.spec.ts:16]: as discusssed, tests have been added

**Review by mstrasinskis — APPROVED** [review]
LGTM

File: frontend/svelte/src/lib/utils/icp.utils.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (mstrasinskis) [reply frontend/svelte/src/lib/utils/icp.utils.ts:36]: No the code is cool, the rule is tricky :D

**Review by mstrasinskis — APPROVED** [review]

**Pull request merged**